DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,269 @@
+
+You need a Linux machine with Docker for development. After you make changes to the code, run the build with:
+
+```bash
+make
+```
+
+To build a Harvester ISO, run:
+
+```bash
+make build-iso
+```
+
+Additionally, see [Build ISO images](https://github.com/harvester/harvester/wiki/Build-OCI-and-ISO-images) for more build options. If you'd like to push the images to a Docker registry, use the following commands:
+
+```bash
+export REPO={docker user name}
+export PUSH=true
+export USE_LOCAL_IMAGES=true
+make
+make build-iso
+```
+
+After building, check the `dist/artifacts` directory for the resulting files. You can test the ISO on physical servers or use [Vagrant](https://github.com/harvester/ipxe-examples/tree/main/vagrant-pxe-harvester) to test on virtual machines.
+
+### Automation Testing
+
+In addition to unit tests in each project, we have a dedicated repository, [harvester/tests](https://github.com/harvester/tests), to test Harvester features. See the [README](https://github.com/harvester/tests/blob/main/README.md) for project setup details.
+
+Use these directories to test different targets:
+- API: [tests/harvester_e2e_tests/apis](https://github.com/harvester/tests/tree/main/harvester_e2e_tests/apis)
+- Integration: [tests/harvester_e2e_tests/integrations](https://github.com/harvester/tests/tree/main/harvester_e2e_tests/integrations)
+
+If you're not sure how to start from scratch, take a look at these two PRs:
+- API: [harvester/tests#2068](https://github.com/harvester/tests/pull/2068)
+- Integration: [harvester/tests#2054](https://github.com/harvester/tests/pull/2054)
+
+Here are example commands to run the tests:
+
+```bash
+# xxx.xxx.xxx.xxx -> This is your cluster IP.
+# Example: run an API test file
+pytest harvester_e2e_tests/apis/test_support_bundle.py --username {account} --password {password} --endpoint {https://xxx.xxx.xxx.xxx}
+
+# Example: run an integration test file
+pytest harvester_e2e_tests/integrations/test_1_images.py --username {account} --password {password} --endpoint {https://xxx.xxx.xxx.xxx}
+```
+
+## Before Opening a Pull Request
+
+Make sure you have an issue on GitHub. All PRs require an issue. Describe the problem you're trying to solve and how you plan to address it. If the issue has the `require/hep` label, a [HEP (Harvester Enhancement Proposal)](./enhancements) is required for discussion before implementation.
+
+The general workflow is:
+- Create or find an issue.
+- Discuss why we need to fix it, what the goal is, and how to solve it.
+- Open a pull request for an HEP if needed.
+- Open a pull request to solve the issue.
+- Test the solution.
+
+### How to Find an Issue to Work On
+
+All issues are tracked in the harvester/harvester repository. Issues with a milestone are ready for development, including the Planning milestone. You can use [this filter](https://github.com/harvester/harvester/issues?q=is%3Aissue%20state%3Aopen%20has%3Amilestone) to find open issues with a milestone.
```

Comment:
We can consider adding a `good-first-issue` label to issues that are suitable for new contributors to work on.